### DISCUSSION TITLE
关于近期New Bing加强了wss连接的限制

### DISCUSSION THREAD

**Discussion by juzeon** [viopsa233/AI-testers-Forum#92] [交流]
众所周知New Bing全程用到的接口基本只有两个。

一个是创建conversation的接口，比如这个：<https://edgeservices.bing.com/edgesvc/turing/conversation/create>。这个接口比较好操作，调用很简单（GET一下即可），基本能访问就能创建。当然前提是带上cookies。

另一个是Websocket的接口，比如`wss://sydney.bing.com`。

这个接口前几天进一步封锁了IP，原来改`X-Forwarded-For`的方法基本失效了。要使用真实的能够解锁的IP才行。

[go-proxy-bingai](https://github.com/adams549659584/go-proxy-bingai)其实很早就解决了这个问题，用的是Cloudflare Workers做反代。[SydneyQt](https://github.com/juzeon/SydneyQt)后来也就延续了这个解决方案。

昨天又进一步加强了限制，即使套了Cloudflare Workers也不行了。

解决方案是请求Websocket的时候，通过extra_headers带上cookies。这个在SydneyQt之前的一个[PR](https://github.com/juzeon/SydneyQt/pull/52)中被解决了，所以不受影响。go-proxy-bingai目前还没有更新。

但按照一般使用Websocket的习惯上，是不会附带自定义的extra_headers的，这点就很奇怪。

包括使用官方网页版也是，即使用已经解锁的IP去访问，如果在未登录状态下，但发送一条消息时，还是会显示“当前网络环境不支持该功能”之类的提示。

目前暂不清楚微软今后是否会继续加强限制。

**Comment by SoraRoseous** [comment]
实际上 @renqabs 在使用的时候发现随便输一个cookie都可以继续正常使用的。。。但是随便输的cookie是画不了图片的，只能聊天，而且聊天的对话次数上限和未登录是一样的10次，而不是使用正确cookie的30次。~可能是微软程序猿偷懒了吧~

**Reply by renqabs** [reply]
不是我发现的，是我在go-proxy-bingai的issue里看到的https://github.com/adams549659584/go-proxy-bingai/issues/347#issuecomment-1637638998

**Reply by juzeon** [reply]
牛的！

**Comment by Ninokz** [comment]
确实貌似
![Screenshot_2023_0718_042733](https://github.com/viopsa233/AI-testers-Forum/assets/33475233/0b65235f-706a-417b-afb9-22a542998764)
